lib/bolt/bolt_option_parser.rb

```diff
@@ -100,258 +142,454 @@ def get_help_text(subcommand, action = nil)
     HELP
 
     PLAN_CONVERT_HELP = <<~HELP
-      Usage: bolt plan convert <plan_path>
+      NAME
+          convert
+
+      USAGE
+          bolt plan convert <path> [options]
 
-      Available options are:
+      DESCRIPTION
+          Convert a YAML plan to a Bolt plan.
+
+          Converting a YAML plan may result in a plan that is syntactically
+          correct but has different behavior. Always verify a converted plan's
+          functionality.
+
+      EXAMPLE
+          bolt plan convert path/to/plan/myplan.yaml
+    HELP
+
+    PLAN_RUN_HELP = <<~HELP
+      NAME
+          run
+
+      USAGE
+          bolt plan run <plan> [parameters] [options]
+
+      DESCRIPTION
+          Run a plan on the specified targets.
+
+      EXAMPLE
+          bolt plan run canary --targets target1,target2 command=hostname
     HELP
 
     PLAN_SHOW_HELP = <<~HELP
-      Usage: bolt plan show <plan>
+      NAME
+          show
 
-      Available actions are:
-        show                             Show list of available plans
-        show <plan>                      Show details for plan
+      USAGE
+          bolt plan show [plan] [options]
 
-      Available options are:
+      DESCRIPTION
+          Show available plans and plan documentation.
+
+          Omitting the name of a plan will display a list of plans available
+          in the Bolt project.
+
+          Providing the name of a plan will display detailed documentation for
+          the plan, including a list of available parameters.
+
+      EXAMPLE
+          Display a list of available tasks
+            bolt plan show
+          Display documentation for the canary task
+            bolt plan show aggregate::count
     HELP
 
-    PLAN_RUN_HELP = <<~HELP
-      Usage: bolt plan run <plan> [parameters]
+    PROJECT_HELP = <<~HELP
+      NAME
+          project
 
-      Parameters are of the form <parameter>=<value>.
+      USAGE
+          bolt project <action> [options]
 
-      #{examples('plan run canary command=hostname', 'run the canary plan on')}
-      Available options are:
+      DESCRIPTION
+          Create and migrate Bolt projects
+
+      ACTIONS
+          init              Create a new Bolt project
+          migrate           Migrate a Bolt project to the latest version
     HELP
 
-    FILE_HELP = <<~HELP
-      Usage: bolt file <action>
+    PROJECT_INIT_HELP = <<~HELP
+      NAME
+          init
+
+      USAGE
+          bolt project init [directory] [options]
+
+      DESCRIPTION
+          Create a new Bolt project.
 
-      Available actions are:
-        upload <src> <dest>              Upload local file or directory <src> to <dest> on each node
+          Specify a directory to create a Bolt project in. Defaults to the
+          curent working directory.
 
-      #{examples('file upload /tmp/source /etc/profile.d/login.sh', 'upload a file to')}
-      Available options are:
+      EXAMPLES
```

Comment:
A couple of these are named `EXAMPLES` and most are `EXAMPLE`. We should be consistent.